android/src/main/java/com/pw/droplet/braintree/Braintree.java

```diff
@@ -1,301 +1,316 @@
-
-                            if (numberError != null) {
-                                errors.put("card_number", numberError.getMessage());
-                            }
-
-                            if (cvvError != null) {
-                                errors.put("cvv", cvvError.getMessage());
-                            }
-
-                            if (expirationDateError != null) {
-                                errors.put("expiration_date", expirationDateError.getMessage());
-                            }
-
-                            // TODO add more fields
-                            if (postalCode != null) {
-                                errors.put("postal_code", postalCode.getMessage());
-                            }
-
-                            nonceErrorCallback(gson.toJson(errors));
-                        } else {
-                            nonceErrorCallback(errorWithResponse.getErrorResponse());
+                        break;
+                    case BraintreeRequestCodes.VENMO:
+                        if (mVenmoClient != null) {
+                            mVenmoClient.onBrowserSwitchResult(
+                                    browserSwitchResult,
+                                    this::handleVenmoResult
+                            );
                         }
-                    }
+                        break;
                 }
-            });
-            this.setToken(token);
-            successCallback.invoke(this.getToken());
+            }
         }
     }
 
-    public void nonceCallback(String nonce) {
-        if (nonce != null) {
-            this.successCallback.invoke(nonce);
-        } else {
-            this.errorCallback.invoke("Sorry, we cannot access you payment method.");
+    @ReactMethod
+    public void setup(final String token) {
+        if (mBraintreeClient == null || !token.equals(mToken)) {
+            mCurrentActivity = (FragmentActivity) getCurrentActivity();
+            mBraintreeClient = new BraintreeClient(mContext, token);
+
+            new DataCollector(mBraintreeClient).collectDeviceData(
+                    mContext,
+                    (result, e) -> mDeviceData = result);
+            mToken = token;
         }
     }
 
-    public void objCallback(WritableMap obj) {
-        this.successCallback.invoke(obj);
-    }
+    @ReactMethod
+    public void paypalRequest(
+            final ReadableMap parameters,
+            final Promise promise
+    ) {
+        mPromise = promise;
+
+        String description = parameters.hasKey("description") ?
+                parameters.getString("description") :
+                "FLUZ APP";
+        String localeCode = parameters.hasKey("localeCode") ?
+                parameters.getString("localeCode") :
+                "US";
+
+        if (mCurrentActivity != null) {
+            mPayPalClient = new PayPalClient(mBraintreeClient);
+            PayPalVaultRequest request = new PayPalVaultRequest();
```

Comment:
Depreciated. need to add hasUserLocationConsent param. 

![image](https://github.com/StronkMan/react-native-braintree-no-ui/assets/165301863/1701e2b2-bd2c-4e33-ab7e-ac9d4a6f5dc5)